### ISSUE TITLE
Lacks support for sql.Out as an argument

### ISSUE THREAD

**Issue by maguirja** [DATA-DOG/go-sqlmock#144]
It seems that all of the code around the arguments is built around the idea of driver.Value as the argument type. Are there  any plans to change that to conform to how it's being done in the built-in package so that types like sql.Out can be supported?

I am getting errors simply when calling a function like ExecContext with a sql.Out argument, so it's not specifically related to how I'm testing, as far as I can discern.

**Comment by l3pp4rd** [comment]
Hi, yes this one could be supported if possible, any chance you are willing to contribute it?

**Comment by theodesp** [comment]
Hey I will try to implement this but I'm new to the code. Any hints?

**Comment by l3pp4rd** [comment]
well, the idea for sql.Out is to update its value during the call on Exec. probably the custom Argument matcher could set that variable when matching the arguments.

**Comment by theodesp** [comment]
So I will add a new Argument Matcher for sql.Out only and at that stage if we match it we should set the value for `result ` to the Out parameter? In that case should we just check if the passed variable has the same result value as with the WillReturnResult?

One other thing. Is it better to do this using a custom ValueConverter instead? Because if we attempt to use the default one it will not recognise the sql.Out Value so we have to add something like:

```
type sqlOutType struct{}

func (sqlOutType) ConvertValue(v interface{}) (driver.Value, error) {
	switch v.(type) {
	case sql.Out:
		return v, nil
	}
	return nil, fmt.Errorf("sqlmock: couldn't convert %v (%T) into type sql.Out", v, v)
}
```
so that the `argsMatches` method would not error when we try to call ConvertValue

Thoughts?

**Comment by l3pp4rd** [comment]
implemented in #174
